src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketsTelemetry.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics.Tracing;
+
+namespace System.Net.Sockets
+{
+    [EventSource(Name = "System.Net.Sockets")]
+    internal sealed class SocketsTelemetry : EventSource
+    {
+        public static readonly SocketsTelemetry Log = new SocketsTelemetry();
+
+        [NonEvent]
+        public void ConnectStart(Internals.SocketAddress address)
+        {
+            if (IsEnabled())
```

Comment:
These IsEnabled checks need to check for the proper event level.  Otherwise, for example, in this case if the EventSource is enabled but not at the Informational level, we're still going to end up paying for the ToString operation.

Reply:
Should we attempt to avoid the ToString here entirely, or is the overhead considered acceptable? If the latter, similar logic in #38409 could be simplified.

Reply:
EventLevel check added.

Reply:
This is called once per connection, so it seems acceptable. Additionally, I checked ToString() implementation and it looks quite efficient.